tempodb/encoding/streaming_block.go

```diff
@@ -25,17 +25,13 @@ type CompactorBlock struct {
 	cfg *BlockConfig
 }
 
-// NewCompactorBlock creates a ... new compactor block!
-func NewCompactorBlock(cfg *BlockConfig, id uuid.UUID, tenantID string, metas []*backend.BlockMeta, estimatedObjects int) (*CompactorBlock, error) {
+// NewStreamingBlock creates a ... new streaming block. Objects are appended one at a time to the backend.
+func NewStreamingBlock(cfg *BlockConfig, id uuid.UUID, tenantID string, metas []*backend.BlockMeta, estimatedObjects int) (*StreamingBlock, error) {
 	if len(metas) == 0 {
 		return nil, fmt.Errorf("empty block meta list")
 	}
 
-	if estimatedObjects <= 0 {
```

Comment:
any particular reason this was removed?

Reply:
There are quite a few tests that create empty WAL/complete blocks and expect it to work.  It used to because CompleteBlock did not enforce at least one trace to be present, even though it also created a bloom filter, because the bloom filter accounts for it here:  https://github.com/grafana/tempo/blob/master/tempodb/encoding/common/bloom.go#L23

Now that it is switched to Compactor/StreamingBlock under the hood, the tests do not pass this check.  Removing this check seems preferable, as the bloom filter will "work" regardless, and the behavior expected by the tests is preserved.